examples/chip-tool/commands/common/ModelCommand.cpp

```diff
@@ -89,10 +91,18 @@ void ModelCommand::SendCommand(ChipDeviceController * dc)
     if (buffer == nullptr)
     {
         ChipLogError(chipTool, "Failed to allocate memory for packet data.");
-        return;
+        return false;
     }
 
     uint16_t dataLength = EncodeCommand(buffer, bufferSize, mEndPointId);
+    if (dataLength == 0)
+    {
+        PacketBuffer::Free(buffer);
+        Command * command = reinterpret_cast<Command *>(this);
```

Comment:
This can be a `static_cast`, I'd think.

But also, the fact that `GetName()` is overridden by `NetworkCommand` is not great.  We should either change `Command` to have `GetCommandName()/mCommandName`, or change the `NetworkCommand` API and member to something other than `GetName/mName`.  Followup, please.

Reply:
+1 ... this is very strange/unusual. Could we maybe do Command::GetName() ?

are we redefining the method without it being virtual I guess? We need separate naming

Reply:
`ChipLogError(chipTool, "Error while encoding data for command: %s", Command::GetName());`

?

Reply:
I renamed the `NetworkCommand::GetName`-> `NetworkCommand::GetNetworkName`